File: AppIntroPractice/app/src/main/java/com/bdpqchen/appintropractice/MainActivity.java
package com.bdpqchen.appintropractice;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    private Context mContext;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.mContext = this;
        setContentView(R.layout.activity_main);

        Thread thread = new Thread(new Runnable() {
            @Override
            public void run() {
//                startActivity(new Intent(mContext, AppIntroActivity.class));
                startActivity(new Intent(mContext, ScreenSlideViewPager.class));
            }
        });
        thread.start();



    }
}

File: MagicImage/app/src/main/java/com/bdpqchen/magicimage/old/MainActivity.kt
package com.bdpqchen.magicimage.old

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import com.bdpqchen.magicimage.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val panorama = findViewById<Button>(R.id.panorama)
        panorama.setOnClickListener({ startAction(PanoramaActivity::class.java) })
        val filter = findViewById<Button>(R.id.filter)
        filter.setOnClickListener({ startAction(FilterActivity::class.java) })



    }


    private fun startAction(clazz: Class<*>) {
        startActivity(Intent(this, clazz))
    }


}
